pulsar-broker/src/test/java/org/apache/pulsar/broker/namespace/OwnerShipForCurrentServerTestBase.java

```diff
@@ -118,96 +86,22 @@ protected void startBroker() throws Exception {
             conf.setWebServicePortTls(Optional.of(0));
             serviceConfigurationList.add(conf);
 
-            PulsarService pulsar = spyWithClassAndConstructorArgs(PulsarService.class, conf);
-
-            setupBrokerMocks(pulsar);
-            pulsar.start();
+            PulsarTestContext.Builder testContextBuilder =
+                    PulsarTestContext.startableBuilder()
+                            .config(conf);
+            if (i > 0) {
+                testContextBuilder.reuseMockBookkeeperAndMetadataStores(pulsarTestContexts.get(0));
```

Comment:
Why are we getting the first one here?

Reply:
PulsarTestContext holds the mock bookkeeper and metadata store (mockzookeeper) references. The "reuseMockBookkeeperAndMetadataStores" can be used to share the mock bookkeeper and zookeeper from another context.